src/index.js

```diff
@@ -52,69 +43,56 @@ export function useVirtual({
 
   const [measuredCache, setMeasuredCache] = React.useState({})
 
-  const { measurements, reversedMeasurements } = React.useMemo(() => {
+  const measurements = React.useMemo(() => {
     const measurements = []
-    const reversedMeasurements = []
-
-    for (let i = 0, j = size - 1; i < size; i++, j--) {
+    for (let i = 0; i < size; i++) {
       const measuredSize = measuredCache[i]
       const start = measurements[i - 1] ? measurements[i - 1].end : paddingStart
       const size =
         typeof measuredSize === 'number' ? measuredSize : estimateSize(i)
       const end = start + size
-      const bounds = { index: i, start, size, end }
-      measurements[i] = {
-        ...bounds,
-      }
-      reversedMeasurements[j] = {
-        ...bounds,
-      }
+      measurements[i] = { index: i, start, size, end }
     }
-    return { measurements, reversedMeasurements }
+    return measurements
   }, [estimateSize, measuredCache, paddingStart, size])
 
   const totalSize = (measurements[size - 1]?.end || 0) + paddingEnd
 
-  let start = React.useMemo(
-    () =>
-      reversedMeasurements.reduce(
-        (last, rowStat) => (rowStat.end >= scrollOffset ? rowStat : last),
-        reversedMeasurements[0]
-      ),
-    [reversedMeasurements, scrollOffset]
-  )
+  Object.assign(latestRef.current, {
+    overscan,
+    measurements,
+    outerSize,
+    totalSize,
+  })
 
-  let end = React.useMemo(
-    () =>
-      measurements.reduce(
-        (last, rowStat) =>
-          rowStat.start <= scrollOffsetPlusOuterSize ? rowStat : last,
-        measurements[0]
-      ),
-    [measurements, scrollOffsetPlusOuterSize]
-  )
+  const [range, setRange] = React.useState({ start: 0, end: 0 })
 
-  let startIndex = start ? start.index : 0
-  let endIndex = end ? end.index : 0
+  useIsomorphicLayoutEffect(() => {
+    const element = parentRef.current
 
-  // Always add at least one overscan item, so focus will work
-  startIndex = Math.max(startIndex - overscan, 0)
-  endIndex = Math.min(endIndex + overscan, size - 1)
+    const onScroll = () => {
+      const scrollOffset = element[scrollKey]
+      latestRef.current.scrollOffset = scrollOffset
+      setRange(prevRange => calculateRange(latestRef.current, prevRange))
+    }
 
-  const latestRef = React.useRef({})
+    // Determine initially visible range
+    onScroll()
 
-  latestRef.current = {
-    measurements,
-    outerSize,
-    scrollOffset,
-    scrollOffsetPlusOuterSize,
-    totalSize,
-  }
+    element.addEventListener('scroll', onScroll, {
+      capture: false,
+      passive: true,
+    })
+
+    return () => {
+      element.removeEventListener('scroll', onScroll)
+    }
+  }, [parentRef, scrollKey, size /* required */])
```

Comment:
I tried this PR, list doesn't corrently rendering. fixed by 
```suggestion
  }, [parentRef.current, scrollKey, size /* required */])
```

Reply:
Good catch! The old `useScroll` hook had some code that handled the parentRef changing which I had omitted. Looks like this simple change should solve that 👍